alpha/lib/model/entry.php

```diff
@@ -342,145 +346,90 @@ public function incViews ( $should_save = true )
+	public function alignFieldValue($field, $value, $update_db)
 	{
-		$defaultLanguage = $this->getPartnerDefaultLanguage();
-		$multiLangMapping = json_decode($this->getMultiLanguageMapping(), true);
-		$isMapUpdated = false;
-		$defaultValue = null;
-		$requestLanguage = kCurrentContext::getLanguage();
-		
-		if($defaultLanguage)
-		{
-			if (is_array($originalValue))
-			{
-				$defaultValue = $this->removeFromMapping($originalValue, $defaultLanguage);
-				if ($field == self::NAME && !$defaultValue && !$multiLangMapping)
-				{
-					throw new kCoreException('Entry name mapping should include default language: ' . $originalValue, KalturaErrors::INVALID_FIELD_VALUE);
-				}
-				$this->setValueByLanguageInMultiLangMapping($multiLangMapping, $field, $originalValue);
-				$this->setMultiLanguageMapping($multiLangMapping ? json_encode($multiLangMapping) : null);
-				$isMapUpdated = true;
-			}
-			elseif ($multiLangMapping && $requestLanguage && ($requestLanguage !== $defaultLanguage))
-			{
-				$this->setValueByLanguageInMultiLangMapping($multiLangMapping, $field, $originalValue, $requestLanguage);
-				$isMapUpdated = true;
-			}
-			elseif($field == self::NAME && $requestLanguage && ($requestLanguage !== $defaultLanguage))
-			{
-				throw new kCoreException('Entry name should be in default language: ' . $originalValue, KalturaErrors::INVALID_FIELD_VALUE);
-			}
-		}
-		if ($defaultValue)
+		$supportedFields = self::getMultiLingualSupportedFields();
+		switch ($field)
 		{
-			$isMapUpdated = false;
-			switch ($field)
-			{
-				case self::NAME:
-					$originalValue = kString::alignUtf8String($defaultValue, self::MAX_NAME_LEN);
-					break;
-				case self::DESCRIPTION:
-					$originalValue = $defaultValue;
-					break;
-				case self::TAGS:
-					$originalValue =  ktagword::updateTags($this->tags, $defaultValue , $update_db );
-					break;
-			}
+			case $supportedFields['name']:
+				return kString::alignUtf8String($value, self::MAX_NAME_LEN);
+			case $supportedFields['description']:
+				return $value;
+			case $supportedFields['tags']:
+				return ktagword::updateTags($this->tags, $value , $update_db );
+			default:
+				return null;
 		}
-		
-		return array(self::IS_DEFAULT => !$isMapUpdated, self::DEFAULT_VALUE => $originalValue);
+	}
+	
+	public function setObjectDefaultLanguage($v)
+	{
+		$this->putInCustomData(self::ENTRY_DEFAULT_LANGUAGE, $v);
+	}
+	
+	public function getObjectDefaultLanguage()
+	{
+		return $this->getFromCustomData(self::ENTRY_DEFAULT_LANGUAGE, null, 'EN');
 	}
 	
 	public function setName($v)
 	{
-		PeerUtils::setExtension($this, $v, self::MAX_NAME_LEN, __FUNCTION__);
-		$multiLingualResult = $this->handleMultiLanguageInput(self::NAME, $v);
-		if ($multiLingualResult[self::IS_DEFAULT])
-		{
-			return parent::setName($multiLingualResult[self::DEFAULT_VALUE]);
-		}
-		return;
+		$multiLingualResult = multiLingualUtils::handleMultiLanguageInput($this, self::NAME, $v);
```

Comment:
since this won't always be multilingual result (meaning we will work with old flow) i think it better have a more generic name for the variable like $name = or $decription =